content/recipes/repl.md

```diff
@@ -0,0 +1,103 @@
+### Read-Eval-Print-Loop (REPL)
+
+REPL is a simple interactive environment that takes single user inputs, executes them, and returns the result to the user.
+REPL feature lets you inspect your dependency graph and call methods on your providers (and controllers) directly from your terminal.
+
+#### Usage
+
+The REPL functionality is enabled when you call `repl` over some root NestJS module like so:
+
+```typescript
+@@filename(repl)
+import { repl } from '@nestjs/core';
+import { AppModule } from './app.module';
+
+async function bootstrap() {
+  await repl(AppModule);
+}
+bootstrap();
+@@switch
+import { repl } from '@nestjs/core';
+import { AppModule } from './app.module';
+
+async function bootstrap() {
+  await repl(AppModule);
+}
+bootstrap();
+```
+
+And then start the app just like the usual:
+
+```bash
+$ npm run start
+```
+
+> info **Hint** `repl` returns a [Node.js REPL server](https://nodejs.org/api/repl.html), thus you can customize it as you wish.
+
+#### Native functions
+
+The built-in NestJS REPL comes with few native functions that are globally available when you start the REPL.
+You can call `help()` to list them out.
+
+If you don't recall what's the signature (ie: expected parameters and return types) of some function, you can call `<function_name>.help`.
+For instance:
+
+```text
+> select.help
+Allows navigating through the modules tree, for example, to pull out a specific instance from the selected module.
+Interface: select(token: DynamicModule | ClassRef) => INestApplicationContext
+```
+
+> info **Hint** Those function interfaces are written in [TypeScript function type expression syntax](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-type-expressions).
+
+#### `debug`
```

Comment:
Instead of adding a separate section per method, we could provide a simple table (similar to other chapters in the docs) with a list of available methods + short description. What would be more beneficial to devs (I believe) is an example use-case (a set of commands showcasing how to access a random service + trigger its method from within the terminal + some screenshots/gifs/videos)

Reply:
I changed to table. The downside is that the interface part won't have syntax highlighting anymore

But I don't know how we could embed images/gifs here :/

Reply:
I didn't mean to add gifs in the table (if that's what you're asking about 😄 )

Reply:
yeah haha

instead of images/gifs what do you think of use https://asciinema.org?

something like

![image](https://user-images.githubusercontent.com/13461315/174633612-7e0405c1-836c-4253-884a-fa3d4ba1010d.png)

I just need to fix the size.

Reply:
I've never used it & would love to see it in action :)